### PR TITLE
SAT ignored fields key for basic read

### LINKED ISSUE #21737: [SAT]  ignored_fields key for basic read test
Sometimes sources change data in records for each read, like a `last_refresh_time ` column.

See https://github.com/airbytehq/airbyte/pull/21419, https://github.com/airbytehq/airbyte/pull/21593 and https://github.com/airbytehq/airbyte/pull/21775 for examples

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*Add `ignored_fields` section for basic read*

## How
- *added new section to config*
- *added bypass reason for empty fields*
- *handled ignored fields in basic read and full refresh tests*
- *updated unittests and added new ones*
- *made configs backward compatible*

### CODE REVIEW

**Comment by YowanR** [comment]
@roman-yermilov-gl @lazebnyi Looks like this is important for Stripe. @airbytehq/connector-operations are you able to help prioritize this review?

File: airbyte-integrations/connectors/source-faker/acceptance-test-config.yml
Comment (erohmensing) [inline airbyte-integrations/connectors/source-faker/acceptance-test-config.yml:34]: Why would this be fast changing?

File: airbyte-integrations/connectors/source-google-pagespeed-insights/acceptance-test-config.yml
Comment (erohmensing) [inline airbyte-integrations/connectors/source-google-pagespeed-insights/acceptance-test-config.yml:42]: This seems legit for `analysisUTCTimestamp`, why is it the case for `lighthouseResult`?

File: airbyte-integrations/connectors/source-zoom/acceptance-test-config.yml
Comment (erohmensing) [inline airbyte-integrations/connectors/source-zoom/acceptance-test-config.yml:45]: Does our test account have fast-changing list of meetings and webinars?

**Review by erohmensing — COMMENTED** [review]
@YowanR I'm concerned about us putting "fast-changing data" as the reason for each ignored column that we don't currently have a commented reason for. Once it gets merged, I don't forsee someone coming back to check them, so I'd like to get it right the first time. Id like some way to confirm that that's the reason. For some of them (parameters like last accessed at, and data like instagram feeds/stories which are populated by some algorithm we don't control, I think it makes sense, but I think there are others that are less clear. 

This is only a review of adding the bypass reasons as a result, I can review the implementation code separately

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:51]:
```suggestion
        invoices:
          - name: invoice_pdf
            bypass_reason: "URL changes upon each request for privacy/security"
          - name: hosted_invoice_url
            bypass_reason: "URL changes upon each request for privacy/security"
```

File: airbyte-integrations/connectors/source-instagram/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-instagram/acceptance-test-config.yml:42]: Which user are we using here?  If this is a fake/test user, this data shouldn't change.  If this is a "real" user, then this information would be changing, but then is inappropriate for this test (not a sandbox)

File: airbyte-integrations/connectors/source-azure-table/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-azure-table/acceptance-test-config.yml:29]: If we really are using a static dataset, this should't be changing?

File: airbyte-integrations/connectors/source-clickup-api/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-clickup-api/acceptance-test-config.yml:21]: why would our list of team members be changing?

File: airbyte-integrations/connectors/source-coinmarketcap/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-coinmarketcap/acceptance-test-config.yml:43]: This source uses real market data, and that's probably always changing... so 👍

File: airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-dockerhub/acceptance-test-config.yml:28]: This seems to imply we are testing against a real docker image that someone's using?

File: airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml
Comment (evantahler) [inline airbyte-integrations/connectors/source-shopify/acceptance-test-config.yml:45]: What causes these properties to change?  Isn't our test storefront/products static?  Does SAT "make a purchase" and change the inventory?

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml
Comment (evantahler) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:51]: For streams that really have changing data, we need to add better reasons to explain why

**Review by evantahler — CHANGES_REQUESTED** [review]
👍 on the SAT change

Like @erohmensing, I'm also worried about how many properties we are skipping here.  I added comments below on the streams I think which shouldn't have changing data... and perhaps this means that we aren't using good sandboxes.

For the remaining bypassed properties, please add more detail to why that specific column is changing and why.  I added an example of this to Stripe.

**Comment by lazebnyi** [comment]
@evantahler @erohmensing @roman-yermilov-gl I'm not sure if we should overwrite the old version format to `ignore_fields` during a full upgrade for all connectors. This change can affect many connectors, which can be in different operational states. And we'll get stuck fixing those connectors or in try to add bypass reasons connectors. 

My proposal is add new version only for high strictness SAT. WDYT?

**Comment by roman-yermilov-gl** [comment]
@lazebnyi @evantahler @erohmensing 
As I see from the source code we are really not overriding old configs. If I override configs, then I also must migrate those configs to new version, otherwise we will have a mix of the old config version with new style `ignored_fields`. It is not allowed (and logically incorrect) to have such a mix of versions in one config so it is better to leave configs as is and make the feature backward compatible. PR is updated

**Comment by evantahler** [comment]
removing my review in favor of the other folks on the Connector Operations tea

**Comment by erohmensing** [comment]
> I'm not sure if we should overwrite the old version format to ignore_fields during a full upgrade for all connectors. This change can affect many connectors, which can be in different operational states. And we'll get stuck fixing those connectors or in try to add bypass reasons. My proposal is add new version only for high strictness SAT. WDYT?

> As I see from the source code we are really not overriding old configs. If I override configs, then I also must migrate those configs to new version, otherwise we will have a mix of the old config version with new style ignored_fields. It is not allowed (and logically incorrect) to have such a mix of versions in one config so it is better to leave configs as is and make the feature backward compatible. PR is updated

@lazebnyi @roman-yermilov-gl As far as I understand it, the proposal is to add the new `ignored_fields` format to multiple types of tests only in the new format, while keeping the old way for the old configs that have not been migrated. Please let me know if my understanding is correct: 
* Connectors that currently use `ignored_fields` with an unmigrated config will be unaffected
* When those connectors try to change their format to the new one (most likely during a switch to high strictness), they will have to implement bypass reasons for their `ignored_fields` (is this enforced in code, or something we should manually enforce?)

In any case, I do think this is a better approach than trying to migrate all at once and this PR being blocked. Let's get this in to unblock the cases that need it and connector ops can be on the look out for configs that don't yet have bypass_reasons when they should (similar to how we now to `allowedHosts`). Maybe we can migrate one connector in this PR so that we can both test the changes and use it as an example?

**Comment by roman-yermilov-gl** [comment]
> > I'm not sure if we should overwrite the old version format to ignore_fields during a full upgrade for all connectors. This change can affect many connectors, which can be in different operational states. And we'll get stuck fixing those connectors or in try to add bypass reasons. My proposal is add new version only for high strictness SAT. WDYT?
> 
> > As I see from the source code we are really not overriding old configs. If I override configs, then I also must migrate those configs to new version, otherwise we will have a mix of the old config version with new style ignored_fields. It is not allowed (and logically incorrect) to have such a mix of versions in one config so it is better to leave configs as is and make the feature backward compatible. PR is updated
> 
> @lazebnyi @roman-yermilov-gl As far as I understand it, the proposal is to add the new `ignored_fields` format to multiple types of tests only in the new format, while keeping the old way for the old configs that have not been migrated. Please let me know if my understanding is correct:
> 
> * Connectors that currently use `ignored_fields` with an unmigrated config will be unaffected
> * When those connectors try to change their format to the new one (most likely during a switch to high strictness), they will have to implement bypass reasons for their `ignored_fields` (is this enforced in code, or something we should manually enforce?)
> 
> In any case, I do think this is a better approach than trying to migrate all at once and this PR being blocked. Let's go ahead and work to get this in get this in to unblock the cases that need it and connector ops can be on the look out for configs that don't yet have bypass_reasons for their ignored_fields when they should (similar to how we now to `allowedHosts`). Maybe we can migrate one connector in this PR so that we can both test the changes and use it as an example?

**- Connectors that currently use ignored_fields with an unmigrated config will be unaffected** - yes

**- When those connectors try to change their format to the new one (most likely during a switch to high strictness), they will have to implement bypass reasons for their ignored_fields (is this enforced in code, or something we should manually enforce?)** - yes, for now there is a special function that reformats a config from old version to new one but it is not called for new configs, so it is required for connector to have `bypass_reason` for `ignored_fields`

**Maybe we can migrate one connector in this PR so that we can both test the changes and use it as an example?** - I tested this feature on the Stripe connector so I already have it there. I can update Stripe PR after this one will be merged

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py
Comment (erohmensing) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py:89]:
```suggestion
        ignored_fields: Optional[Mapping[str, List[IgnoredFieldsConfiguration]]] = {},
```
Do we need to add a default either here or in the method? If it's optional I wonder what will happen if its not declared, since we used to set a default [here](https://github.com/airbytehq/airbyte/pull/22996/files?diff=unified&w=0#diff-82c6340d6fd6cc2a860e7a6080d0eabf830e4e4504a3a10b6f5ccf08818be126L93)

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (erohmensing) [inline airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py:287]:
```suggestion
            {"aa": 23}, 
            {}, 
            does_not_raise()
```
Nit: just for clarity/consistency

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (erohmensing) [inline airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py:330]: These are great for understanding the behavior 👍🏻

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (erohmensing) [inline airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py:355]: I'm not sure if I understand why this fails for the exact order one. Shouldn't it fail because the values don't match? An exact order test would be if we had multiple records output for the stream, which match the initial ones but are out of order

**Review by erohmensing — CHANGES_REQUESTED** [review]
This is looking pretty good! For reference, what I was concerned about was with regards to how we were approaching enforcing bypass reasons (or not). Looks like we have taken Serhii's suggestion of only doing so for high strictness, which I think makes sense. Then we can fix those that are already high strictness and don't have bypass reasons (as they'll start failing), and handle the others as we migrate them to high strictness. This is similar to how we approach bypass reasons for expected_records. 

Most of my comments are just questions or suggestions - the current blocking items are 
* Change to a minor bump instead of a patch
* Implement unit tests for the fixture 
* Making sure we continue to handle ignored_fields not being set in test_sequential_reads

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py:357]: I need this test to ensure that my code works in `if exact_order:` branch

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:901]: I just applied an action that I only needed

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py:204]: done

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/config.py:120]: It has the same field structure for both test configs, so I thought I can declare it one time

File: airbyte-integrations/bases/connector-acceptance-test/Dockerfile
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/Dockerfile]: done

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py:89]: I assigned default value in the fixture

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/compare.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/compare.py]: done

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (roman-yermilov-gl) [reply airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py]: done

File: docs/connector-development/testing-connectors/connector-acceptance-tests-reference.md
Comment (roman-yermilov-gl) [reply docs/connector-development/testing-connectors/connector-acceptance-tests-reference.md:154]: done

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py
Comment (erohmensing) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_full_refresh.py:89]: Thanks, I saw that - you were a step ahead of me while I was reviewing 😄

File: airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py
Comment (erohmensing) [reply airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py:357]: Got it - I misinterpreted this as a new test for the exact_order behavior

File: airbyte-integrations/bases/connector-acceptance-test/CHANGELOG.md
Comment (erohmensing) [inline airbyte-integrations/bases/connector-acceptance-test/CHANGELOG.md:3]:
```suggestion
## 0.7.0
```

**Review by erohmensing — APPROVED** [review]
Ready to see this in action in Stripe! Updated the changelog, should be ready for publishing

**Review by evantahler — APPROVED** [review]

**Comment by roman-yermilov-gl** [comment]
/publish connector=bases/connector-acceptance-test auto-bump-version=false

**Comment by roman-yermilov-gl** [comment]
Publish was successful except updating publish comment in PR: `Error: API rate limit exceeded for installation ID 6913263.`

**Pull request merged**
